In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


use for give path to working directory

In [3]:
!pwd

/content


Give a path to connect with our file


In [4]:
%cd /content/drive/MyDrive/datasets/housing

/content/drive/MyDrive/datasets/housing


import pandas & numpy

In [5]:
import pandas as pd
import numpy as np

In [ ]:
#raw_df = pd.read_csv('datasets/Hyderabad.csv')

In [6]:
raw_df = pd.read_csv('Hyderabad.csv')

In [7]:
raw_df

Price  Area     Location  No. of Bedrooms  Resale  MaintenanceStaff  \
0      6968000  1340     Nizampet                2       0                 0   
1     29000000  3498  Hitech City                4       0                 0   
2      6590000  1318    Manikonda                2       0                 0   
3      5739000  1295        Alwal                3       1                 0   
4      5679000  1145   Kukatpally                2       0                 0   
...        ...   ...          ...              ...     ...               ...   
2513  11000000  1460     Nacharam                2       1                 9   
2514  26000000  1314    Manikonda                2       1                 9   
2515  13300000  2625     Madhapur                3       1                 9   
2516  10800000  2050  Hitech City                3       0                 9   
2517  10400000  1805     Narsingi                3       0                 9   

      Gymnasium  SwimmingPool  LandscapedGardens  JoggingTrack  ...  \
0             1             1                  1             1  ...   
1             1             1                  1             1  ...   
2             1             0                  0             0  ...   
3             0             0                  0             0  ...   
4             0             0                  1             0  ...   
...         ...           ...                ...           ...  ...   
2513          9             9                  9             9  ...   
2514          9             9                  9             9  ...   
2515          9             9                  9             9  ...   
2516          9             9                  9             9  ...   
2517          9             9                  9             9  ...   

      LiftAvailable  BED  VaastuCompliant  Microwave  GolfCourse  TV  \
0                 1    0                1          0           0   0   
1                 1    0                1          0           0   0   
2                 0    0                0          0           0   0   
3                 1    0                0          0           0   0   
4                 1    0                0          0           0   0   
...             ...  ...              ...        ...         ...  ..   
2513              9    9                9          9           9   9   
2514              9    9                9          9           9   9   
2515              9    9                9          9           9   9   
2516              9    9                9          9           9   9   
2517              9    9                9          9           9   9   

      DiningTable  Sofa  Wardrobe  Refrigerator  
0               0     0         0             0  
1               0     0         0             0  
2               0     0         0             0  
3               0     0         0             0  
4               0     0         0             0  
...           ...   ...       ...           ...  
2513            9     9         9             9  
2514            9     9         9             9  
2515            9     9         9             9  
2516            9     9         9             9  
2517            9     9         9             9  

[2518 rows x 40 columns]

it is use for total count of the data
total entries = 2518

In [ ]:
raw_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2518 entries, 0 to 2517
Data columns (total 40 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Price                2518 non-null   int64 
 1   Area                 2518 non-null   int64 
 2   Location             2518 non-null   object
 3   No. of Bedrooms      2518 non-null   int64 
 4   Resale               2518 non-null   int64 
 5   MaintenanceStaff     2518 non-null   int64 
 6   Gymnasium            2518 non-null   int64 
 7   SwimmingPool         2518 non-null   int64 
 8   LandscapedGardens    2518 non-null   int64 
 9   JoggingTrack         2518 non-null   int64 
 10  RainWaterHarvesting  2518 non-null   int64 
 11  IndoorGames          2518 non-null   int64 
 12  ShoppingMall         2518 non-null   int64 
 13  Intercom             2518 non-null   int64 
 14  SportsFacility       2518 non-null   int64 
 15  ATM                  2518 non-null   int64 
 16  ClubHo

In [ ]:
raw_df.shape

(2518, 40)

Number of columns = 40

In [ ]:
raw_df.columns.shape

(40,)

In [ ]:
raw_df.columns

Index(['Price', 'Area', 'Location', 'No. of Bedrooms', 'Resale',
       'MaintenanceStaff', 'Gymnasium', 'SwimmingPool', 'LandscapedGardens',
       'JoggingTrack', 'RainWaterHarvesting', 'IndoorGames', 'ShoppingMall',
       'Intercom', 'SportsFacility', 'ATM', 'ClubHouse', 'School',
       '24X7Security', 'PowerBackup', 'CarParking', 'StaffQuarter',
       'Cafeteria', 'MultipurposeRoom', 'Hospital', 'WashingMachine',
       'Gasconnection', 'AC', 'Wifi', 'Children'splayarea', 'LiftAvailable',
       'BED', 'VaastuCompliant', 'Microwave', 'GolfCourse', 'TV',
       'DiningTable', 'Sofa', 'Wardrobe', 'Refrigerator'],
      dtype='object')

show for 29th row data 

In [8]:
raw_df.loc[29]

Price                     9800000
Area                         1520
Location               Gachibowli
No. of Bedrooms                 2
Resale                          1
MaintenanceStaff                0
Gymnasium                       0
SwimmingPool                    0
LandscapedGardens               0
JoggingTrack                    0
RainWaterHarvesting             0
IndoorGames                     0
ShoppingMall                    0
Intercom                        0
SportsFacility                  0
ATM                             0
ClubHouse                       0
School                          0
24X7Security                    0
PowerBackup                     1
CarParking                      0
StaffQuarter                    0
Cafeteria                       0
MultipurposeRoom                0
Hospital                        0
WashingMachine                  0
Gasconnection                   0
AC                              0
Wifi                            0
Children'splay

In [ ]:
raw_df[raw_df['No. of Bedrooms']>=4]

Price  Area       Location  No. of Bedrooms  Resale  \
1     29000000  3498    Hitech City                4       0   
21    29000000  3498    Hitech City                4       0   
46    30000000  2956  Banjara Hills                4       1   
103   28200000  3500    Hitech City                4       1   
106   17700000  2729     Gachibowli                4       0   
...        ...   ...            ...              ...     ...   
2445   3000000  2300      Manikonda                4       1   
2464  12600000  4320     Gachibowli                5       1   
2466   6500000  3600    Toli Chowki                4       1   
2471   9057000  4000  Himayat Nagar                5       1   
2499   5500000  1860       Nacharam                4       1   

      MaintenanceStaff  Gymnasium  SwimmingPool  LandscapedGardens  \
1                    0          1             1                  1   
21                   0          1             1                  1   
46                   0          0             0                  0   
103                  0          1             1                  1   
106                  0          1             1                  1   
...                ...        ...           ...                ...   
2445                 9          9             9                  9   
2464                 9          9             9                  9   
2466                 9          9             9                  9   
2471                 9          9             9                  9   
2499                 9          9             9                  9   

      JoggingTrack  RainWaterHarvesting  IndoorGames  ShoppingMall  Intercom  \
1                1                    1            1             0         1   
21               1                    1            1             0         1   
46               0                    0            0             0         0   
103              1                    1            1             0         1   
106              0                    0            1             0         0   
...            ...                  ...          ...           ...       ...   
2445             9                    9            9             9         9   
2464             9                    9            9             9         9   
2466             9                    9            9             9         9   
2471             9                    9            9             9         9   
2499             9                    9            9             9         9   

      SportsFacility  ATM  ClubHouse  School  24X7Security  PowerBackup  \
1                  0    1          1       0             1            1   
21                 0    1          1       0             1            1   
46                 0    0          0       0             0            0   
103                0    1          1       0             1            1   
106                0    1          0       0             0            0   
...              ...  ...        ...     ...           ...          ...   
2445               9    9          9       9             9            9   
2464               9    9          9       9             9            9   
2466               9    9          9       9             9            9   
2471               9    9          9       9             9            9   
2499               9    9          9       9             9            9   

      CarParking  StaffQuarter  Cafeteria  MultipurposeRoom  Hospital  \
1              1             1          0                 1         0   
21             1             1          0                 1         0   
46             0             0          0                 0         0   
103            1             1          0                 1         0   
106            0             0          0                 0         0   
...          ...           ...        ...               ...       ...   
2445           9          

In [ ]:
print(raw_df['No. of Bedrooms']>=4)

0       False
1        True
2       False
3       False
4       False
        ...  
2513    False
2514    False
2515    False
2516    False
2517    False
Name: No. of Bedrooms, Length: 2518, dtype: bool


In [ ]:
features=['Area', 'Location', 'No. of Bedrooms',  'Price']
raw_df=raw_df[features]
raw_df


Area     Location  No. of Bedrooms     Price
0     1340     Nizampet                2   6968000
1     3498  Hitech City                4  29000000
2     1318    Manikonda                2   6590000
3     1295        Alwal                3   5739000
4     1145   Kukatpally                2   5679000
...    ...          ...              ...       ...
2513  1460     Nacharam                2  11000000
2514  1314    Manikonda                2  26000000
2515  2625     Madhapur                3  13300000
2516  2050  Hitech City                3  10800000
2517  1805     Narsingi                3  10400000

[2518 rows x 4 columns]

In [ ]:
raw_df.Area>100

0       True
1       True
2       True
3       True
4       True
        ... 
2513    True
2514    True
2515    True
2516    True
2517    True
Name: Area, Length: 2518, dtype: bool

Standard Deviation

In [ ]:
raw_df.describe()

Area  No. of Bedrooms         Price
count  2518.000000      2518.000000  2.518000e+03
mean   1644.571088         2.623511  9.818380e+06
std     746.256956         0.687330  8.777113e+06
min     500.000000         1.000000  2.000000e+06
25%    1160.000000         2.000000  4.760000e+06
50%    1500.000000         3.000000  7.754000e+06
75%    1829.000000         3.000000  1.090000e+07
max    9400.000000         8.000000  1.650000e+08

In [ ]:
raw_df[["Price", "Area"]].describe()

Price         Area
count  2.518000e+03  2518.000000
mean   9.818380e+06  1644.571088
std    8.777113e+06   746.256956
min    2.000000e+06   500.000000
25%    4.760000e+06  1160.000000
50%    7.754000e+06  1500.000000
75%    1.090000e+07  1829.000000
max    1.650000e+08  9400.000000

In [ ]:
raw_df.describe(include='all')

Area    Location  No. of Bedrooms         Price
count   2518.000000        2518      2518.000000  2.518000e+03
unique          NaN         243              NaN           NaN
top             NaN  Kukatpally              NaN           NaN
freq            NaN         166              NaN           NaN
mean    1644.571088         NaN         2.623511  9.818380e+06
std      746.256956         NaN         0.687330  8.777113e+06
min      500.000000         NaN         1.000000  2.000000e+06
25%     1160.000000         NaN         2.000000  4.760000e+06
50%     1500.000000         NaN         3.000000  7.754000e+06
75%     1829.000000         NaN         3.000000  1.090000e+07
max     9400.000000         NaN         8.000000  1.650000e+08

In [ ]:
raw_df.describe(include=[object])

Location
count         2518
unique         243
top     Kukatpally
freq           166

In [ ]:
raw_df[(raw_df['Price']>1000000) & (raw_df['No. of Bedrooms']>3)].reset_index(drop=True)

Area       Location  No. of Bedrooms     Price
0    3498    Hitech City                4  29000000
1    3498    Hitech City                4  29000000
2    2956  Banjara Hills                4  30000000
3    3500    Hitech City                4  28200000
4    2729     Gachibowli                4  17700000
..    ...            ...              ...       ...
179  2300      Manikonda                4   3000000
180  4320     Gachibowli                5  12600000
181  3600    Toli Chowki                4   6500000
182  4000  Himayat Nagar                5   9057000
183  1860       Nacharam                4   5500000

[184 rows x 4 columns]

In [ ]:
raw_df.Price.mean()

9818379.636616362